### DISCUSSION TITLE
generic sweeps must start at origin?

### DISCUSSION THREAD

**Discussion by tve** [sgenoud/replicad#97] [Q&A]
Trying to do a sweep along a curve starting anywhere other than the origin produces weird results. Here's a screen shot showing the bezier I'm sweeping and the resulting sweep:

![image](https://github.com/sgenoud/replicad/assets/39480/6badcba1-35e5-4b88-b12c-d98a2c78ea35)

See [code](https://studio.replicad.xyz/workbench?code=UEsDBAoAAAAIAPcuA1cfyvHQEQEAALQBAAAHAAAAY29kZS5qc22QQW6DMBBF95xixAoqStNUyiJSN%252B0BEiWLqkUsHDOAFTNGY1NUoty9BhLURSXLHn8%252F%252F%252Fkaacg6YHgFxlYrKYogKDuSThmCRiiKYrgEAHLiTsMIpo044xsOCvm942%252BMMg8AZI%252FrVQJ%252B%252BTNPbtLL%252F9JIPq9yr%252BRxsNgXSnj%252FzXKXU7eCRX8wHRVYHFA6QZXGaP3g4QTu29MmTu0Znax3tNeCMAo%252Fv8I%252F1rZvJ7MKCVnJY4%252FYRlMgmfaKcQ7HqbAWm5PGD69F2WnI4%252FnlAqVhiXs2pdJ4bBXhztWmMiS0cj9bcNwhXGe4FNqir6b%252BjK5jguxmY2vR4tZPMgESja9COc4wvP9dCJ94QewY1yPjvILrL1BLAQIUAAoAAAAIAPcuA1cfyvHQEQEAALQBAAAHAAAAAAAAAAAAAAAAAAAAAABjb2RlLmpzUEsFBgAAAAABAAEANQAAADYBAAAAAA%253D%253D)

```
const r = replicad

function main() {
  const bz = r.makeBezierCurve([
    [-20, 0, -20],
    [-30, 0, -20],
    [-30, 20, 10]
  ])

  const dia = 6
  const c = r.drawRoundedRectangle(2*dia, dia, dia/6).sketchOnPlane("YZ")

  const swp = r.genericSweep(
    c.wire,
    r.assembleWire([bz]),
    { forceProfileSpineOthogonality: true },
    false
  )

  return [
    { shape: bz, name: "curve" },
    { shape: swp, name: "sweep" }
  ]
}
```

**Comment by sgenoud** [comment]
I am not sure about this. I have found the same thing than you - not sure if there is a better way. This comes directly from open cascade. 

I have tried to stick to sweeping from the origin to obtain the result that I want (or use the `withContact`option). Is there a reason you want the spine to be defined outside of the shape you are sweeping?

There is a need to explore a bit more good ways to use sweeping (and with it an API where common things are easy to do and work out of the box).
